akka-stream/src/main/scala/akka/stream/scaladsl/Flow.scala

```diff
@@ -792,8 +792,8 @@ trait FlowOps[+Out, +Mat] {
    */
   def groupedWithin(n: Int, d: FiniteDuration): Repr[immutable.Seq[Out]] = {
     require(n > 0, "n must be greater than 0")
-    require(d > Duration.Zero)
-    via(new GroupedWithin[Out](n, d).withAttributes(name("groupedWithin")))
+    require(d > Duration.Zero) // FIXME shouldn't the requirements of GroupedWithin be verified *inside* GroupedWithin?
```

Comment:
No, it should be done in the constructor of GroupedWithin, which happen “here” as well.